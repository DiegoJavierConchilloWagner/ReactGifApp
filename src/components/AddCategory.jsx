import React, { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleInputSubmit = (e) => {
        e.preventDefault();

        console.log('handleSubmit', inputValue);

        if( inputValue.trim().length > 2 ){
            setCategories( cats => [ inputValue, ...cats ]);
            setInputValue('')
        }

        
    }

    return (
        <form onSubmit={handleInputSubmit} >
            <p> { inputValue }</p>
            <input  
                type="text" 
                value={inputValue} 
                placeholder="Buscar meme..."
                onChange={ handleInputChange } 
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}