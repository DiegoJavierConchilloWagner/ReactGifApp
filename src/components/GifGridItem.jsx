import React from 'react'
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url }) => {

    return (
        <div className="gifCard animate__animated animate__bounceInUp">
            <img src={ url } alt={ title } />
            <p> { title } </p> 
        </div>

    )
}

GifGridItem.prototypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}