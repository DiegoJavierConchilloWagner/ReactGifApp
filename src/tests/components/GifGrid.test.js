import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {
    
    const category = 'Search';

    test('Debe de mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes UseFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://example.com',
            title: 'Cualquier cosa'
        },
        {
            id: 'ABC1',
            url: 'http://example.com',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs ,
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
    
});
