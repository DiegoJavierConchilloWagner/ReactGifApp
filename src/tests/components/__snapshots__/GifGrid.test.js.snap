// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe de mostrar items cuando se cargan imagenes UseFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Search
     
  </h3>
  <div
    class="loader"
  />
  <div
    className="gifCardContainer"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://example.com"
    />
    <GifGridItem
      id="ABC1"
      key="ABC1"
      title="Cualquier cosa"
      url="http://example.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
     
    Search
     
  </h3>
  <div
    class="loader"
  />
  <div
    className="gifCardContainer"
  />
</Fragment>
`;
